app/controllers/sessions_controller.rb

```diff
@@ -6,7 +6,7 @@ def create
     user = User.find_by(email: session_params[:email])
     if user && user.authenticate(session_params[:password])
       session[:user_id] = user.id
-      if current_user.roles.take.name == "admin"
+      if current_user.roles.name == "admin"
```

Comment:
@cmacaulay  current_user.roles.name returns the name of the class which bypasses this check. 
current_user.roles.name returns "Role"
"Role" != "admin"
I'll rework this logic to make it easier to check and avoid future nil returns. Thanks for highlighting the issue.